src/tests/core/blocks/test_create_block.c

```diff
@@ -0,0 +1,208 @@
+  );
+  mu_assert(
+    "Coinbase num Inputs is not 0",
+    tx->num_inputs == 0
+  );
+  mu_assert(
+    "Coinbase num Outputs is not 1",
+    tx->num_outputs == 1
+  );
+  mu_assert(
+    "Coinbase output amount is not what is expected",
+    tx->outputs[0].amt == test+BLOCK_REWARD
+  );
+  return NULL;
+}
+
+static char *test_get_txs_from_mempool(){
+  _fill_mempool();
+  Transaction **test_ptr = malloc(sizeof(Transaction**));
+  unsigned int num_tx = get_txs_from_mempool(&test_ptr);
+  mu_assert(
+    "Wrong number of transactions",
+    num_tx == 2
+  );
+  mu_assert(
+    "Coinbase TX num_input doesn't match",
+    test_ptr[0]->num_inputs == 0
+  );
+  mu_assert(
+    "Coinbase TX num_outputs doesn't match",
+    test_ptr[0]->num_outputs == 1
+  );
+  mu_assert(
+    "Coinbase TX output amt doesn't match",
+    test_ptr[0]->outputs[0].amt == 10+BLOCK_REWARD
+  );
+  mu_assert(
+    "TX num_input doesn't match",
+    test_ptr[1]->num_inputs == 1
+  );
+  mu_assert(
+    "TX num_outputs doesn't match",
+    test_ptr[1]->num_outputs == 1
+  );
+  mu_assert(
+    "TX output amt doesn't match",
+    test_ptr[1]->outputs[0].amt == 90
+  );
+  unsigned char *total_hash = hash_all_tx(test_ptr,2);
+  unsigned char *total_hash2 = hash_all_tx(test_ptr,2);
+  mu_assert(
+    "Total hash is inconsistent",
+    memcmp(total_hash, total_hash2, ALL_TX_HASH_LEN) == 0
+  );
+
+  BlockHeader *test_header = create_block_header(test_ptr, 2);
+  mu_assert(
+    "Total hash is not consistent in header creation",
+    memcmp(total_hash, test_header->all_tx, ALL_TX_HASH_LEN) == 0
+  );
+  mu_assert(
+    "Header nonce invalid",
+    test_header->nonce == 0
+  );
+  mu_assert(
+    "Header timestamp invalid",
+    test_header->timestamp > 1647573975
+  );
+
+
+  Block *test_block = create_block_alloc();
+  mu_assert(
+    "Block txs coinbase is bad",
+    test_block->txs[0]->num_inputs !=1
+  );
+  mu_assert(
+    "TX output amt doesn't match",
+    test_block->txs[1]->outputs[0].amt == 90
+  );
+  change_nonce(test_block);
+  mu_assert(
+    "Nonce increment failed",
+    test_block->header.nonce == 1
+  );
+  unsigned char header_hash[BLOCK_HASH_LEN];
+  hash_blockheader(header_hash, &(test_block->header));
+  unsigned char header_hash2[BLOCK_HASH_LEN];
+  hash_blockheader(header_hash2, &(test_block->header));
+  mu_assert(
+    "Header hash is inconsistent",
+    memcmp(header_hash, header_hash2, BLOCK_HASH_LEN) == 0
+  );
+  mu_assert(
+    "Header hash failed",
+    try_header_hash(&(test_block->header)) == 1
+  );
+  mu_assert(
+    "Block hash failed",
+    mine_block(test_block) == 1
+  );
+
+  while(mine_block(test_block) != 0){
+    change_nonce(test_block);
+  }
+  print_block_header(&(test_block->header), "");
+  hash_blockheader(header_hash, &(test_block->header));
+  dump_buf("", "MINED TX: ", header_hash, BLOCK_HASH_LEN);
+  free(test_block);
+  free(test_header);
+  free(total_hash);
+  free(total_hash2);
+  free(test_ptr);
+  return NULL;
+}
```

Comment:
This seems like it should be 5 different tests (get_txs, hash_all, create_header, create_block, mine)

Reply:
Yeah you are probably right, will pull the setup bit's into another helper and split into tests